client/src/components/AuditBoard/BallotAudit.tsx

```diff
@@ -226,14 +229,37 @@ const BallotAuditContest = ({
         ...interpretation,
         interpretation: checked ? value : null,
         choiceIds: [],
+        hasInvalidWriteIn: false,
+      })
+    } else if (value === INVALID_WRITE_IN) {
+      let newInterpretation: Interpretation | null
+      if (interpretation.choiceIds.length > 0) {
+        newInterpretation = Interpretation.VOTE
+      } else if (checked) {
+        newInterpretation = Interpretation.BLANK
+      } else {
+        newInterpretation = null
+      }
+      setInterpretation({
+        ...interpretation,
+        interpretation: newInterpretation,
+        hasInvalidWriteIn: checked,
       })
     } else {
       const choiceIds = checked
         ? [...interpretation.choiceIds, value]
         : interpretation.choiceIds.filter(v => v !== value)
+      let newInterpretation: Interpretation | null
+      if (choiceIds.length > 0) {
+        newInterpretation = Interpretation.VOTE
+      } else if (interpretation.hasInvalidWriteIn) {
+        newInterpretation = Interpretation.BLANK
+      } else {
+        newInterpretation = null
+      }
       setInterpretation({
         ...interpretation,
-        interpretation: choiceIds.length > 0 ? Interpretation.VOTE : null,
+        interpretation: newInterpretation,
```

Comment:
This logic is getting a bit complicated. I wonder if it would be simpler to store in React state a direct representation of the checkboxes and then derive an interpretation from it.

Reply:
Good call - let me see if I can tweak 🛠️